power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -777,25 +781,27 @@ class TapPositionOptimizerImpl<std::tuple<TransformerTypes...>, StateCalculator,
             }
         }
 
-        IntS search(bool prefer_higher = true) const {
+        IntS search(bool prefer_higher_) const {
             // This logic is used to determin which of the middle points could be of interest
             // given strategy used in optimization:
             // Since in BinarySearch we insist on absolute upper and lower bounds, we only need to
             // find the corresponding mid point. std::midpoint returns always the lower mid point
             // if the range is of even length. This is why we need to adjust bounds accordingly.
             // Not because upper bound and lower bound might be reversed, which is not possible.
+            bool const prefer_higher = control_at_tap_side_ ? !prefer_higher_ : prefer_higher_;
             auto const primary_bound = prefer_higher ? upper_bound_ : lower_bound_;
             auto const secondary_bound = prefer_higher ? lower_bound_ : upper_bound_;
```

Comment:
this logic was originally introduced to swap upper and lower bound. Maybe it's a good idea to make the `control_at_tap_side` the new default, e.g.

```suggestion
            bool const prefer_lower = control_at_tap_side_ == prefer_higher_;
            auto const primary_bound = prefer_lower ? lower_bound_ : upper_bound_ ;
            auto const secondary_bound = prefer_lower ? upper_bound_ : lower_bound_ ;
```

or otherwise explicitly mention

```suggestion
            bool const inverted_bounds = control_at_tap_side_ != prefer_higher_;
            auto const primary_bound = inverted_bounds ? upper_bound_ : lower_bound_;
            auto const secondary_bound = inverted_bounds ? lower_bound_ : upper_bound_;
```

Reply:
This suggestion is making less sense to me and appears to be a matter of opinion.